### PR TITLE
Remove C casts in w3dwater code

### PR DESCRIPTION
**Pull request by xezon**

### CODE REVIEW

File: src/game/client/water/w3dwater.cpp [hunk 1]
```diff
@@ -639,11 +641,11 @@ int WaterRenderObjClass::Init(float water_level, float dx, float dy, SceneClass
     m_shaderClass = g_zFillAlphaShader;
     m_shaderClass.Set_Cull_Mode(ShaderClass::CULL_MODE_DISABLE);
     m_alphaClippingTexture = W3DAssetManager::Get_Instance()->Get_Texture("TSMoonLarg.tga");
-    m_skyBox = ((GameAssetManager *)W3DAssetManager::Get_Instance())
+    m_skyBox = static_cast<GameAssetManager *>(W3DAssetManager::Get_Instance())
```
Comment (duncanspumpkin) [inline src/game/client/water/w3dwater.cpp:644]: Is there not a method that gets the specific type?

File: src/game/client/water/w3dwater.cpp [hunk 2]
```diff
@@ -1069,7 +1072,7 @@ bool WaterRenderObjClass::Get_Clipped_Water_Plane(CameraClass *cam, AABoxClass *
 
 inline unsigned long F2DW(float f)
 {
-    return *((unsigned *)&f);
+    return *(reinterpret_cast<unsigned long *>(&f));
```
Comment (duncanspumpkin) [inline src/game/client/water/w3dwater.cpp:1075]: uint32_t suggest also changing the function signature to uint32_t

**Review by duncanspumpkin — APPROVED** [review]
Looks okay. Is there not a method that gets a GameAssetManager directly from the global. Perhaps I'm misremembering

File: src/game/client/water/w3dwater.cpp (hunk 1 above)
Comment (xezon) [reply src/game/client/water/w3dwater.cpp:644]: I don't understand.

File: src/game/client/water/w3dwater.cpp (hunk 2 above)
Comment (xezon) [reply src/game/client/water/w3dwater.cpp:1075]: Yeah good point. There is another review with a 2 more of these functions. I think that should be done in a new review entirely.

File: src/game/client/water/w3dwater.cpp (hunk 1 above)
Comment (xezon) [reply src/game/client/water/w3dwater.cpp:644]: Hmm no. `static W3DAssetManager *&s_theInstance;` is protected class member.

**Pull request merged**